From 0140ea7d4868ed0e80a2915470f5ebbca36d4a93 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 14 Aug 2017 13:21:33 +0300
Subject: [PATCH 1215/1345] net: mvpp2x: add 2.5G SGMII force link support

commit  75d8040e16668505fccca53848644effa3c3f6dd from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Enable MV_EMAC_F_SGMII2_5 flag if 2500 speed is set.

Change-Id: I73cccedce490d5cfabbf399db8a837bf1680ec68
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42972
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index d710c74..5037e1d 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -4666,6 +4666,8 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 		fixed_link_node = of_get_child_by_name(emac_node, "fixed-link");
 		port->mac_data.duplex = of_property_read_bool(fixed_link_node,
 				"full-duplex");
+		if (port->mac_data.speed == SPEED_2500)
+			port->mac_data.flags |= MV_EMAC_F_SGMII2_5;
 		if (of_property_read_u32(fixed_link_node, "speed",
 					 &port->mac_data.speed))
 			return -EINVAL;
-- 
1.7.9.5

